secondarylib/src/main/scala/com/socrata/datacoordinator/secondary/SecondaryWatcher.scala

```diff
@@ -107,12 +107,19 @@ class SecondaryWatcher[CT, CV](universe: => Managed[SecondaryWatcher.UniverseTyp
           // Prefer cookie handed-back by the exception, but since we write notes in out cookies currently that we don't
           // want to lose, let's avoid overriding those if we are not given Some other cookie string
           manifest(u).markSecondaryDatasetBroken(job, cookie.orElse(job.initialCookie))
-        case rlse@ReplayLaterSecondaryException(reason, cookie) =>
-          if (job.replayNum < maxReplays) {
-            val replayAfter = Math.min(replayWait.toSeconds * Math.log(job.replayNum + 2), maxReplayWait.toSeconds)
+        case rlse@ReplayLaterSecondaryException(reason, cookie, delay) =>
+          val (replayDelay, replayNum) =
+            delay match {
+              case Some(d) => (d, if (job.replayNum >= maxReplays) 0 else job.replayNum)
```

Comment:
Does this mean that if a delay is set it will never stop attempting to replay? Is that desired?

Reply:
yes, the control is in the hand of the secondary plugin if that parameter is used.  We can put a bound in the hand of secondary-watcher too.  It just becomes complicated.  There is no rush to do/solve this problem.  The current way to disable read/write works.  It is just a little cumbersome.